http-filter-example/http_filter_v2_integration_test.cc

```diff
@@ -0,0 +1,82 @@
+#include "common/grpc/codec.h"
+#include "common/grpc/common.h"
+#include "api/filter/network/http_connection_manager.pb.h"
+
+#include "test/integration/http_integration.h"
+#include "test/integration/utility.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+class HttpFilterSamplev2IntegrationTest : public HttpIntegrationTest,
+                                        public testing::TestWithParam<Network::Address::IpVersion> {
+public:
+  HttpFilterSamplev2IntegrationTest() : HttpIntegrationTest(Http::CodecClient::Type::HTTP1, GetParam()) {}
+  /**
+   * Initializer for an individual integration test.
+   */
+ void SetUp() override {
+    HttpIntegrationTest::SetUp();
+    initialize();
+  }
+
+  void createUpstreams() override {
+    HttpIntegrationTest::createUpstreams();
+    fake_upstreams_.emplace_back(new FakeUpstream(0, FakeHttpConnection::Type::HTTP2, version_));
+    ports_.push_back(fake_upstreams_.back()->localAddress()->ip()->port());
+  }
+
+  void initialize() override {
+    config_helper_.addFilter(
+        "{ name: sample, config: {} }");
+    config_helper_.addConfigModifier([](envoy::api::v2::Bootstrap& bootstrap) {
+      auto* sample_cluster = bootstrap.mutable_static_resources()->add_clusters();
+      sample_cluster->MergeFrom(bootstrap.static_resources().clusters()[0]);
+      sample_cluster->set_name("service1");
+      sample_cluster->mutable_http2_protocol_options();
```

Comment:
Why do you setup an additional service here? The basic config already has a cluster https://github.com/envoyproxy/envoy/blob/master/test/config/utility.cc#L41.

Reply:
Initially I did not setup. But with out that it fails with assertion at https://github.com/envoyproxy/envoy/blob/master/test/config/utility.cc#L110. That is probably because of the way the for loop is setup at 
https://github.com/envoyproxy/envoy/blob/master/test/config/utility.cc#L100??. So I have added this.